org.adempiere.base/src/org/compiere/util/CCache.java

```diff
@@ -130,7 +155,11 @@ public CCache (String tableName, String name, int initialCapacity, int expireMin
 		m_name = name;
 		m_tableName = tableName;
 		setExpireMinutes(expireMinutes);
-		m_maxSize = maxSize; 
+		int propMaxSize = getCacheMaxSize(name);
+		if (propMaxSize >= 0)
+			m_maxSize = propMaxSize;
+		else
+			m_maxSize = maxSize; 

```

Comment:
instead of ignoring the maxSize method parameter, I think we should override `public CCache (String tableName, String name, int initialCapacity, int expireMinutes, boolean distributed)` instead and uses either maxSize from getCacheMaxSize(name) or CacheMgt.MAX_SIZE (if getCacheMaxSize(name) return -1).

Reply:
Hi @hengsin - I don't understand your suggestion, can you please clarify?

Something good about the suggested approach here is that we can change **any** cache size without needing to change the code to use a different method.